sim/midas/src/main/cc/core/simif.cc

```diff
@@ -1,14 +1,15 @@
 // See LICENSE for license details.
 
 #include "simif.h"
+#include "core/config.h"
 #include "core/simulation.h"
 #include "core/stream_engine.h"
 
 #include <iostream>
 
 simif_t::simif_t(const TargetConfig &config,
                  const std::vector<std::string> &args)
-    : config(config), args(args) {
+    : config(config), args(args), registry() {
   for (auto &arg : args) {
```

Comment:
In the spirit of thinning out simif, you should move this into the metasim layer since it's only used there (separate PR is fine).

Reply:
Yes, I had that in mind after the header rewrite is merged.